bin/findHalfcoverage.py

```diff
@@ -0,0 +1,161 @@
+
+	return scaf_lc
+
+
+def get_cov_peaks(file):
+# example is depthgraph.txt, "coverage\tbasepair count"
+
+	myPeakFile = open(file, "r")
+
+	rows = []
+	for line in myPeakFile:
+		line = line.replace("\n", "")
+		items = re.split("\t", line)
+		rows.append(items)
+
+	myPeakFile.close()
+	#print(rows[0])
+	peakCov = sorted(rows, key=lambda cov:int(cov[1]), reverse=1)[0][0]
+
+	if int(peakCov) == 0:
+		peakCov = sorted(rows, key=lambda cov:int(cov[1]), reverse=1)[1][0]
+
+
+
+	halfPeak = int(peakCov)/2
+	qrtPeak  = int(peakCov)/4
+
+	print("#Coverage Peak is %s, HalfPeak is %s, QuarterPeak is %s " % (peakCov, halfPeak, qrtPeak) )
+
+
+	return(peakCov, halfPeak, qrtPeak)
+
+
+def calc_coverage(scafsize, totallowcov) :
+# calculate the % for lowcov coverage over entire scaffold.
+
+	return totallowcov/scafsize*100
+
+
+def getArguments():
+# get indivudual arguments from user
+
+	parser = OptionParser(version="%prog 1.0")
+	parser.add_option("-c", "--coveragefile", 
+                  action="store", type="string", dest="covfile",
+                  help="Scaffold Coverage filename")
+	parser.add_option("-m", "--mygenome", 
+                  action="store", type="string", dest="mygenome", 
+                  help="mygenome file, scaffold - size file")
+	parser.add_option("-d", "--depthgraph", 
+                  action="store", type="string", dest="depth", 
+                  help="depthgraph file, bp count at each depth")
+	parser.add_option("-w", "--wiggle", 
+                  action="store", type="string", dest="wig", 
+                  help="wiggle room to add to depth cutoff ie 30X + wiggleroom.  Default is 5X")
+	parser.add_option("--cut", 
+                  action="store", type="string", dest="covcut", 
+                  help="%Number for coverage cutoff to include in results.  ie 50% of scaffold needs to be under diploid peak etc.  Default is 60%")	
+
+	(options, args) = parser.parse_args()
+
+	if (options.covfile == None or options.mygenome == None or options.depth == None):
+		print("Missing Options")
+		exit()
+
+	cov_filename        = options.covfile
+	mygenome_filename   = options.mygenome
+	depth_filename      = options.depth
+	wiggle_opt          = options.wig
+	cutoff_opt          = options.covcut
+
+	return (cov_filename, mygenome_filename, depth_filename, wiggle_opt, cutoff_opt)
+
+
+def main():
+# main program	
+
+	(arg1, arg2, arg3, arg4, arg5) = getArguments()
+
+	if arg4 == None:
+		arg4 = 5
+
+	if arg5 == None:
+		arg5 = 60
+			
+
+	scaffold_sizes = load_scafsize(arg2)
+	(hapCov, dipCov, tetCov) = get_cov_peaks(arg3)
+	scaffold_lowcovsum = getTotallength_undercov(arg1, dipCov, arg4)
+
+	for scaffoldName in scaffold_lowcovsum:
+		if (scaffoldName == ""):
+			continue
+
+		#print("==" + scaffoldName)
+		totalSize = float(scaffold_sizes[scaffoldName])
+		lowcovSize = float(scaffold_lowcovsum[scaffoldName])
+
+		coverage = calc_coverage(totalSize, lowcovSize)
+
+		if (coverage > arg5):
+
+			if (totalSize >250000):
```

Comment:
Make this a command-line parameter (which may have 250000 as the default value) so that you can easily modify it from the pipeline